File: 121. Best Time to Buy and Sell Stock.cpp
#include <climits>
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if (prices.size() == 0) return 0;
        int maxx = -INT_MAX;
        int end = 0;
        int start = prices[0];
        for (int i = 1; i < prices.size(); i++)
        {
            end = prices[i];
            maxx = max(maxx, end - start);
            start = min(start, end);
        }
        return maxx > 0 ? maxx : 0;
    }
};

File: 121. Best Time to Buy and Sell Stock.py
import sys
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        if not prices:
            return 0

        maxx = -sys.maxint
        start = prices[0]
        end = 0
        for i in range(1, len(prices)):
            end = prices[i]
            maxx = max(maxx, end - start)
            start = min(start, end)
        return 0 if maxx < 0 else maxx
